# Imports

In [23]:
from tqdm import tqdm_notebook as tqdm
import matplotlib
import time
import csv

In [24]:
%matplotlib inline

import matplotlib
import numpy as np
import matplotlib.pyplot as plt

# Sample Hourly Data

## Count

In [31]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [28]:
count = 0
for i in csv_reader:
    count += 1
    
print count

2141026


from one set of hourly data we have 2,141,026 entries. This is more than enough data and the rest of it while cool, is not necessary to get a good start

## Field Names

In [29]:
csv_reader.fieldnames

['WBAN',
 'Date',
 'Time',
 'StationType',
 'SkyCondition',
 'SkyConditionFlag',
 'Visibility',
 'VisibilityFlag',
 'WeatherType',
 'WeatherTypeFlag',
 'DryBulbFarenheit',
 'DryBulbFarenheitFlag',
 'DryBulbCelsius',
 'DryBulbCelsiusFlag',
 'WetBulbFarenheit',
 'WetBulbFarenheitFlag',
 'WetBulbCelsius',
 'WetBulbCelsiusFlag',
 'DewPointFarenheit',
 'DewPointFarenheitFlag',
 'DewPointCelsius',
 'DewPointCelsiusFlag',
 'RelativeHumidity',
 'RelativeHumidityFlag',
 'WindSpeed',
 'WindSpeedFlag',
 'WindDirection',
 'WindDirectionFlag',
 'ValueForWindCharacter',
 'ValueForWindCharacterFlag',
 'StationPressure',
 'StationPressureFlag',
 'PressureTendency',
 'PressureTendencyFlag',
 'PressureChange',
 'PressureChangeFlag',
 'SeaLevelPressure',
 'SeaLevelPressureFlag',
 'RecordType',
 'RecordTypeFlag',
 'HourlyPrecip',
 'HourlyPrecipFlag',
 'Altimeter',
 'AltimeterFlag']

## Sample Entry

In [33]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))
csv_reader.next()

{'Altimeter': '30.25',
 'AltimeterFlag': ' ',
 'Date': '20070501',
 'DewPointCelsius': '-1.0',
 'DewPointCelsiusFlag': ' ',
 'DewPointFarenheit': '30',
 'DewPointFarenheitFlag': ' ',
 'DryBulbCelsius': '7.0',
 'DryBulbCelsiusFlag': ' ',
 'DryBulbFarenheit': '45',
 'DryBulbFarenheitFlag': ' ',
 'HourlyPrecip': ' ',
 'HourlyPrecipFlag': ' ',
 'PressureChange': ' ',
 'PressureChangeFlag': ' ',
 'PressureTendency': ' ',
 'PressureTendencyFlag': ' ',
 'RecordType': 'AA',
 'RecordTypeFlag': ' ',
 'RelativeHumidity': ' 56',
 'RelativeHumidityFlag': ' ',
 'SeaLevelPressure': 'M',
 'SeaLevelPressureFlag': ' ',
 'SkyCondition': 'CLR',
 'SkyConditionFlag': ' ',
 'StationPressure': '21.57',
 'StationPressureFlag': ' ',
 'StationType': '0',
 'Time': '0050',
 'ValueForWindCharacter': ' ',
 'ValueForWindCharacterFlag': ' ',
 'Visibility': '10.00',
 'VisibilityFlag': ' ',
 'WBAN': '03011',
 'WeatherType': ' ',
 'WeatherTypeFlag': ' ',
 'WetBulbCelsius': '3.3',
 'WetBulbCelsiusFlag': ' ',
 'WetBulbFare